src/nvim/message.c

```diff
@@ -1450,6 +1450,9 @@ void msg_prt_line(char_u *s, int list)
       } else if (c == ' ' && trail != NULL && s > trail) {
         c = lcs_trail;
         attr = hl_attr(HLF_8);
+      } else if (c == ' ' && list && lcs_space != NUL) {
```

Comment:
I like `'\0'` because there's almost no chance that it will be mixed up with `NULL`, unlike `NUL`, but in practice the two rarely seem to get mixed up. #652 discusses it, and while I still like `'\0'` more, I'm not sure everyone else agrees.